src/source_distribution.rs

```diff
@@ -456,7 +456,7 @@ fn add_cargo_package_files_to_sdist(
             let mut deps_to_keep = known_path_deps.clone();
             // Also need to the main Python binding crate
             let main_member_name = abs_manifest_dir
-                .strip_prefix(&sdist_root)
+                .file_name()
```

Comment:
What if `pyo3-mixed-workspace-py` is located at `rust/python/pyo3-mixed-workspace`? Then in workspace `Cargo.toml`, `members` would be

```toml
members = [
    "pyo3-mixed-workspace",
    "python/pyo3-mixed-workspace",
]
```

Only use the file name would be wrong? I think we maybe it should be changed to `strip_prefix(&workspace_root)`?

Reply:
Yes, that's a great point. I just updated to strip the `workspace_root` prefix and updated the test to reflect the nested case you described and it seems to be working.